### PR TITLE
[Docs] How to add product variants by options to the cart in Sylius API?

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| License         | MIT

I will open another PR with a tutorial on how to add the product to cart,

### CODE REVIEW

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          "@id": "/api/v2/shop/product-variant-translation/579960",
+          "@type": "ProductVariantTranslation",
+          "id": 579960,
+          "name": "S Petite",
+          "locale": "en_US"
+        }
+      },
+      "price": 7693
+
+And with this information you can add chosen ``product variant`` to cart.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:84]:
```suggestion
And with this information, you can add the chosen ``product variant`` to the cart.

```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
+Add to cart product chosen by product options
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:1]: How to add product variants to the cart in Sylius API?

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+=============================================
+
+In Sylius API to add product to the cart you need fill body request with ``productCode`` and ``productVariantCode``
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:4]:
```suggestion
In  order to add a product variant to the cart in the Sylius API, you need to fill the "add to cart" request's body with ``productCode`` and ``productVariantCode``
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+You can find this data in body of response which is received after the request below is sent.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:5]:
```suggestion
You can find this data in the body of the response which is received after the request below is sent.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: bash
+
+    curl -X GET "https://master.demo.sylius.com/api/v2/shop/products?page=1&itemsPerPage=30" -H  "accept: application/ld+json"
+
+But in real shops the most common case is choosing ``product variant`` by product options.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:11]:
```suggestion
But the most common case is choosing a product variant by its product options.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+So below you can find all requests needed to get ``product variant`` with chosen product options values.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:12]:
```suggestion
So below you can find all requests needed to get the product variant of a product with chosen product options values.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+In this example we want to add to cart `small` and `petite` dress.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:13]:
```suggestion
In this example, we will be adding an "S" (in size) and "petite" (in height) "Beige strappy summer dress" to the cart.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+Firstly you need to get details of product:
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:15]:
```suggestion
Firstly you need to get details of the "Beige strappy summer dress" product by its code:
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: bash
+
+    curl -X GET "https://master.demo.sylius.com/api/v2/shop/products/Beige_strappy_summer_dress" -H  "accept: application/ld+json"
+
+Now you can see that this dress has 2 ``product options``:
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:21]:
```suggestion
In the response you will see that this product has 2 product options:
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: json
+
+    "options": [
+        "/api/v2/shop/product-options/dress_size",
+        "/api/v2/shop/product-options/dress_height"
+      ],
+
+With this data you can check all available ``product option values`` for each ``product option``:
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:30]:
```suggestion
With this data you can check all the available product option values for each product option:
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: bash
+    curl -X GET "https://master.demo.sylius.com/api/v2/shop/product-options/dress_size" -H  "accept: application/ld+json"
+
+And available option:
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:35]:
```suggestion
The response with all the available option values would be:
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+    "values": [
+        "/api/v2/shop/product-option-values/dress_s",
+        "/api/v2/shop/product-option-values/dress_m",
+        "/api/v2/shop/product-option-values/dress_l",
+        "/api/v2/shop/product-option-values/dress_xl",
+        "/api/v2/shop/product-option-values/dress_xxl"
+      ],
+
+In the same way you can check values for ``dress_height``.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:47]:
```suggestion
In the same way, you can check values for the other option - ``dress_height``.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+Now, with all necessary data you can find ``Beige strappy summer dress`` `small` and `petite` variant
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:49]:
```suggestion
Now, with all necessary data, you can find the "Beige strappy summer dress" product's "small" and "petite" variant
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+You need to call get on ``ProductVariant`` collection with parameters: `productName` and with all options value
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:50]:
```suggestion
You need to call a GET on the product variants collection with parameters: ``productName`` and with the chosen option values:
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+.. code-block:: bash
+    curl -X GET "https://master.demo.sylius.com/api/v2/shop/product-variants?product=%2Fapi%2Fv2%2Fshop%2Fproducts%2FBeige_strappy_summer_dress&
+optionValues%5B%5D=%2Fapi%2Fv2%2Fshop%2Fproduct-option-values%2Fdress_height_petite&
+optionValues%5B%5D=%2Fapi%2Fv2%2Fshop%2Fproduct-option-values%2Fdress_s&page=1&itemsPerPage=30" -H  "accept: application/ld+json"
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:55]:
```suggestion
optionValues%5B%5D=%2Fapi%2Fv2%2Fshop%2Fproduct-option-values%2Fdress_s&page=1" -H  "accept: application/ld+json"
```
is this part of the request required?

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+.. warning::
+
+    When you add only part of configured ``product option value`` to parameters in response you may get collection with more than single object.
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:59]:
```suggestion
    When you search by only some of the product's option values in the response you may get a collection with more than one object.
```

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+
+In response you will get collection with only one item:
```
Comment (CoderMaggie) [inline docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:61]:
```suggestion
In the response you should get a collection with only one item:
```

**Review by CoderMaggie — CHANGES_REQUESTED** [review]
As it is a cookbook, why isn't it added here?
https://docs.sylius.com/en/latest/cookbook/index.html#api

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst (hunk 2 above)
Comment (lchrusciel) [reply docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:1]: How to add product variants by options to the cart in Sylius API?

File: docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+        "/api/v2/shop/product-option-values/dress_xxl"
+      ],
+
+In the same way you can check values for ``dress_height``.
+
+Now, with all necessary data you can find ``Beige strappy summer dress`` `small` and `petite` variant
+You need to call get on ``ProductVariant`` collection with parameters: `productName` and with all options value
+
+.. code-block:: bash
+    curl -X GET "https://master.demo.sylius.com/api/v2/shop/product-variants?product=%2Fapi%2Fv2%2Fshop%2Fproducts%2FBeige_strappy_summer_dress&
```
[the remark this answers is not in the document]
Comment (AdamKasp) [reply docs/api/unified_api/add_to_cart_product_chosen_by_product_options.rst:53]: TBH I am not sure, this `curl` was generated by swagger, and I passed it here.

**Review by arti0090 — APPROVED** [review]

**Comment by AdamKasp** [comment]
> As it is a cookbook, why isn't it added here?
> https://docs.sylius.com/en/latest/cookbook/index.html#api

IMO it isn't a cookbook, but an instruction/tutorial, so this place fits better.

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Adam! :tada:

**Pull request merged**